src/main/java/bio/terra/service/tabulardata/azure/StorageTableDao.java

```diff
@@ -0,0 +1,131 @@
+package bio.terra.service.tabulardata.azure;
+
+import bio.terra.model.BulkLoadFileState;
+import bio.terra.model.BulkLoadHistoryModel;
+import com.azure.data.tables.TableClient;
+import com.azure.data.tables.TableServiceClient;
+import com.azure.data.tables.models.ListEntitiesOptions;
+import com.azure.data.tables.models.TableEntity;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class StorageTableDao {
+
+  private static final String LOAD_HISTORY_TABLE_NAME_SUFFIX = "LoadHistory";
+  private static final int AZURE_STORAGE_TABLE_NAME_MAX_LENGTH = 63;
+  private static final int TABLE_NAME_MAX_LENGTH_BEFORE_PREFIX =
+      AZURE_STORAGE_TABLE_NAME_MAX_LENGTH - LOAD_HISTORY_TABLE_NAME_SUFFIX.length();
+
+  /**
+   * Store the results of a bulk file load in an Azure Storage Table
+   *
+   * <p>The table name will be the result of the dataset name passed through {@link
+   * StorageTableDao#azurfyTableName} Entities will be partitioned on the {@param loadTag} and their
+   * row keys will be the value of{@link BulkLoadHistoryModel#getFileId()}
+   *
+   * @param serviceClient A service client for the dataset
+   * @param datasetName the name of the dataset
+   * @param loadTag Load tag to partition on
+   * @param loadTime The time the load occurred
+   * @param loadHistoryArray The models to store
+   */
+  public void loadHistoryToAStorageTable(
+      TableServiceClient serviceClient,
+      String datasetName,
+      String loadTag,
+      Instant loadTime,
+      List<BulkLoadHistoryModel> loadHistoryArray) {
+    var tableName = azurfyTableName(datasetName);
+    TableClient client = serviceClient.createTableIfNotExists(tableName);
+    loadHistoryArray.stream()
+        .map(model -> bulkFileLoadModelToStorageTableEntity(model, loadTag, loadTime))
+        .forEach(client::createEntity);
+  }
+
+  /**
+   * Get the load history results from storage tables
+   *
+   * @param tableServiceClient A client for the dataset
+   * @param datasetName The dataset name
+   * @param loadTag The load tag of the file load
+   * @param offset Results will be offset by this much
+   * @param limit Results will be limited to this many
+   * @return The load history for a load tag within the confines of offset and limit.
+   */
+  public List<BulkLoadHistoryModel> getLoadHistory(
+      TableServiceClient tableServiceClient,
+      String datasetName,
+      String loadTag,
+      int offset,
+      int limit) {
+    var tableClient = tableServiceClient.getTableClient(azurfyTableName(datasetName));
+    ListEntitiesOptions options =
+        new ListEntitiesOptions().setFilter(String.format("PartitionKey eq '%s'", loadTag));
```

Comment:
Is this a possible code injection point? What if `loadTag` contains a single quote that ends the string expression early?

Reply:
Since this is essentially a GET request with URL parameters, base64 encoding the partition key should block any string termination.